src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountingExtractCollectorBatchBuilder.java

```diff
@@ -282,44 +321,147 @@ protected String buildTransactionDescription(ConcurStandardAccountingExtractDeta
                 .toString();
     }
 
-    protected void compareLineNonKeyFieldsAgainstEntry(String itemKey,
-            ConcurStandardAccountingExtractDetailLine saeLine, OriginEntryFull originEntry) {
-        if (!StringUtils.equals(originEntry.getDocumentNumber(), buildDocumentNumber(saeLine))) {
-            LOG.warn("Found a documentNumber/reportId mismatch on at least one line for item key: " + itemKey);
-        } else if (!StringUtils.equals(originEntry.getTransactionLedgerEntryDescription(), buildTransactionDescription(saeLine))) {
-            LOG.warn("Found a lastName/firstName/endDate mismatch on at least one line for item key: " + itemKey);
-        }
+    protected void reportCashAdvance(ConcurStandardAccountingExtractDetailLine saeLine) {
+        updateStatisticFromLineData(ConcurStandardAccountingExtractBatchReportData::getCashAdvancesRelatedToExpenseReports, saeLine);
+    }
+
+    protected void reportCorporateCardPayment(ConcurStandardAccountingExtractDetailLine saeLine) {
+        updateStatisticFromLineData(ConcurStandardAccountingExtractBatchReportData::getExpensesPaidOnCorporateCard, saeLine);
+    }
+
+    protected void reportBypassOfLineWithPseudoPaymentCode(ConcurStandardAccountingExtractDetailLine saeLine) {
+        updateStatisticFromLineData(ConcurStandardAccountingExtractBatchReportData::getTransactionsBypassed, saeLine);
+    }
+
+    protected void updateStatisticFromLineData(
+            Function<ConcurStandardAccountingExtractBatchReportData,ConcurBatchReportSummaryItem> summaryItemGetter,
+            ConcurStandardAccountingExtractDetailLine saeLine) {
+        ConcurBatchReportSummaryItem summaryItem = summaryItemGetter.apply(reportData);
+        summaryItem.setRecordCount(summaryItem.getRecordCount() + 1);
+        summaryItem.setDollarAmount(summaryItem.getDollarAmount().add(saeLine.getJournalAmount().abs()));
+    }
+
+    protected void reportPendingClientLine(ConcurStandardAccountingExtractDetailLine saeLine) {
+        ConcurBatchReportMissingObjectCodeItem pendingClientItem = new ConcurBatchReportMissingObjectCodeItem(

```

Comment:
Nancy and I had discussed this recently, and I think we're okay with potential duplication for now, and we can clean this up as a future enhancement. Also, there may be non-CASH rows that have the "Pending Client" code.